backend/src/serverless/integrations/usecases/discourse/getCategories.ts

```diff
@@ -0,0 +1,30 @@
+import axios, { AxiosRequestConfig } from 'axios'
+import type { DiscourseConnectionParams } from '../../types/discourseTypes'
+import { Logger } from '../../../../utils/logging'
+import { DiscourseCategoryResponse } from '../../types/discourseTypes'
+
+export const getDiscourseCategories = async (
+  params: DiscourseConnectionParams,
+  logger: Logger,
+): Promise<DiscourseCategoryResponse> => {
+  logger.info({
+    message: 'Fetching categories from Discourse',
+    forumHostName: params.forumHostname,
+  })
+  const config: AxiosRequestConfig<any> = {
+    method: 'get',
+    url: `${params.forumHostname}/categories.json`,
```

Comment:
not sure about discourse API but if any endpoint supports pagination we should be using it to prevent too much RAM being used to load ALL the posts/comments/whatever into the memory. I saw pagination being used only in `getTopics.ts` file so just a reminder to check if it can be used somewhere else as well.